app/src/main/java/com/cmput301f20t21/bookfriends/ui/add/AddEditViewModel.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2020. Book Friends Team
+ * All rights reserved.
+ * github URL: https://github.com/CMPUT301F20T21/Book_Friends
+ */
+
+package com.cmput301f20t21.bookfriends.ui.add;
+
+import android.net.Uri;
+
+import androidx.annotation.Nullable;
+import androidx.lifecycle.ViewModel;
+
+import com.cmput301f20t21.bookfriends.enums.BOOK_ERROR;
+import com.cmput301f20t21.bookfriends.services.AuthService;
+import com.cmput301f20t21.bookfriends.services.BookService;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.storage.FirebaseStorage;
+import com.google.firebase.storage.StorageReference;
+
+public class AddEditViewModel extends ViewModel {
+    public interface OnGetImageSuccessCallback {
+        void run(Uri imageUri);
+    }
+
+    public interface OnSuccessCallback {
+        void run();
+    }
+
+    /** called by handlers when async request failed */
+    public interface OnFailCallback {
+        void run(BOOK_ERROR error);
+    }
+
+    private final AuthService authService = AuthService.getInstance();
+    private final BookService bookService = BookService.getInstance();
+    private final StorageReference imageStorageReference = FirebaseStorage.getInstance().getReference();
```

Comment:
I think since photo is related to book, we can hide this info in the bookService as well.